js/bundle.go

```diff
@@ -330,10 +330,27 @@ func (b *Bundle) instantiate(logger logrus.FieldLogger, rt *goja.Runtime, init *
 	init.moduleVUImpl.ctx = context.Background()
 	init.moduleVUImpl.initEnv = initenv
 	init.moduleVUImpl.eventLoop = eventloop.New(init.moduleVUImpl)
+	pgm := init.programs[b.Filename.String()] // this just initializes the program
+	pgm.pgm = b.Program
+	pgm.src = b.Source
+	exports := rt.NewObject()
+	pgm.module = rt.NewObject()
+	_ = pgm.module.Set("exports", exports)
+	init.programs[b.Filename.String()] = pgm
+
 	err = common.RunWithPanicCatching(logger, rt, func() error {
 		return init.moduleVUImpl.eventLoop.Start(func() error {
-			_, errRun := rt.RunProgram(b.Program)
-			return errRun
+			f, errRun := rt.RunProgram(b.Program)
+			if errRun != nil {
+				return errRun
+			}
+			if call, ok := goja.AssertFunction(f); ok {
+				if _, errRun = call(exports, pgm.module, exports); errRun != nil {
+					return errRun
+				}
+				return nil
+			}
+			panic("we shouldn't be able to get here")
```

Comment:
In the rare case, we will have a bug and we will get here I guess we should have a better message.

Reply:
[3e52cb8](https://github.com/grafana/k6/pull/2571/commits/3e52cb8ca3c6a0a9e359d28ff13e8d03ada8d349)